projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts

```diff
@@ -79,30 +82,35 @@ export class AcaMobileAppSwitcherService {
   identifyBrowserAndSetRedirectURL(): void {
     const ua: string = navigator.userAgent.toLowerCase();
     const isAndroid: boolean = ua.indexOf('android') > -1;
-    const isIOS: boolean = ua.indexOf('iphone') > -1 || ua.indexOf('ipad') > -1 || ua.indexOf('ipod') > -1;
+    const isIPadInSafari = /Macintosh/i.test(ua) && navigator.maxTouchPoints && navigator.maxTouchPoints > 1;
+    const isIOS: boolean = ua.indexOf('iphone') > -1 || ua.indexOf('ipad') > -1 || ua.indexOf('ipod') > -1 || isIPadInSafari;
     const currentUrl: string = this.getCurrentUrl();
 
     if (isIOS === true) {
       this.redirectUrl = this.mobileAppSwitchConfig.iphoneUrl + currentUrl;
+      this.appStoreUrl = this.mobileAppSwitchConfig.appStoreUrl;
     } else if (isAndroid === true) {
       this.redirectUrl = this.mobileAppSwitchConfig.androidUrlPart1 + currentUrl + this.mobileAppSwitchConfig.androidUrlPart2;
     }
 
     if (this.redirectUrl !== undefined && this.redirectUrl !== null) {
-      this.openDialog(this.redirectUrl);
+      this.openDialog(this.redirectUrl, this.appStoreUrl);
     }
   }
 
-  openDialog(redirectUrl: string): void {
-    this.dialog.open(OpenInAppComponent, {
-      data: {
-        redirectUrl
-      },
-      hasBackdrop: false,
-      width: 'auto',
-      role: 'dialog',
-      position: { bottom: '20px' }
-    });
+  openDialog(redirectUrl: string, appStoreUrl?: string): void {
+    if (this.dialogRef === null || this.dialogRef === undefined) {
```

Comment:
probably can be reduced to just single condition: 
if (!this.dialogRef)

Reply:
reduced to single condition now